applications/TrilinosApplication/tests/cpp_tests/trilinos_cpp_test_utilities.cpp

```diff
@@ -233,9 +233,12 @@ void TrilinosCPPTestUtilities::CheckSparseMatrixFromLocalMatrix(
     values.reserve(total_size);
     for (std::size_t i = 0; i < rB.size1(); ++i) {
         for (std::size_t j = 0; j < rB.size2(); ++j) {
-            row_indexes.push_back(i);
-            column_indexes.push_back(j);
-            values.push_back(rB(i, j));
+            const double value = rB(i, j);
```

Comment:
why not qdding if It is. a zero?

Reply:
Usually what defines a sparse matrix is the number of non-zero terms, that's why

Reply:
I know that a sparse matrix can have non zero terms which value is actually is zero. The implementation is to check the actual non zero values, otherwise the graph must be provided. I added an additional method where you provide the indices but the values are check from a local matrix, this will work for you?

Reply:
@RiccardoRossi do you agree?

Reply:
Hello @RiccardoRossi

Reply:
mmm, the local matrix type is dense? 

at least give a more descriptive name instead of tolerance. something like "neglibible_value_threshold" or smthg of the sort